### PR TITLE
add unit test for v1.configmap and remove replicate code

### PR DESCRIPTION
**Pull request by yanxuean**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by yanxuean** [comment]
@liggitt PTAL
The commit is migrated from #55465. The only difference is the package name.  
one for pkg\api\pod ,  one for pkg/api/v1/pod.
At first I don't know that they are similar. thanks.

**Comment by yanxuean** [comment]
@liggitt   PTAL

**Comment by yanxuean** [comment]
@liggitt PTAL

**Comment by yanxuean** [comment]
@liggitt  PTAL

File: pkg/api/v1/pod/util_test.go [hunk 1]
```diff
@@ -316,38 +316,113 @@ func TestPodSecrets(t *testing.T) {
 	}
 }
 
-// collectSecretPaths traverses the object, computing all the struct paths that lead to fields with "secret" in the name.
-func collectSecretPaths(t *testing.T, path *field.Path, name string, tp reflect.Type) sets.String {
-	secretPaths := sets.NewString()
+// collectResourcePaths traverses the object, computing all the struct paths that lead to fields with resourcename in the name.
+func collectResourcePaths(t *testing.T, resourcename string, path *field.Path, name string, tp reflect.Type) sets.String {
```
Comment (liggitt) [inline pkg/api/v1/pod/util_test.go]: this function and package should stay in sync with `pkg/api/pod/util.go`

Comment (yanxuean) [reply pkg/api/v1/pod/util_test.go]: They is same.  
The function in pkg/api/pod/util.go have been modified in pr https://github.com/kubernetes/kubernetes/pull/55465.

**Comment by yanxuean** [comment]
@liggitt   PTAL

File: pkg/api/v1/pod/util.go [hunk 2]
```diff
@@ -194,7 +194,7 @@ func visitContainerConfigmapNames(container *v1.Container, visitor Visitor) bool
 }
 
 // GetContainerStatus extracts the status of container "name" from "statuses".
-// It also returns if "name" exists.
```
Comment (liggitt) [inline pkg/api/v1/pod/util.go]: the existing doc is more correct... it is describing what the boolean value represents

**Comment by liggitt** [comment]
nit on doc. lgtm otherwise

File: pkg/api/v1/pod/util.go (hunk 2 above)
Comment (yanxuean) [reply pkg/api/v1/pod/util.go]: done

**Pull request merged**